src/Compilers/Core/Portable/InternalUtilities/StringTable.cs

```diff
@@ -146,72 +145,10 @@ internal string Add(ReadOnlySpan<char> chars)
         }
 
         internal string Add(string chars, int start, int len)
-        {
-            var hashCode = Hash.GetFNVHashCode(chars, start, len);
-
-            // capture array to avoid extra range checks
-            var arr = _localTable;
-            var idx = LocalIdxFromHash(hashCode);
-
-            var text = arr[idx].Text;
-
-            if (text != null && arr[idx].HashCode == hashCode)
-            {
-                var result = arr[idx].Text;
-                if (StringTable.TextEquals(result, chars, start, len))
-                {
-                    return result;
-                }
-            }
-
-            string? shared = FindSharedEntry(chars, start, len, hashCode);
-            if (shared != null)
-            {
-                // PERF: the following code does element-wise assignment of a struct
-                //       because current JIT produces better code compared to
-                //       arr[idx] = new Entry(...)
-                arr[idx].HashCode = hashCode;
-                arr[idx].Text = shared;
-
-                return shared;
-            }
-
-            return AddItem(chars, start, len, hashCode);
-        }
-
-        internal string Add(char chars)
-        {
-            var hashCode = Hash.GetFNVHashCode(chars);
-
-            // capture array to avoid extra range checks
-            var arr = _localTable;
-            var idx = LocalIdxFromHash(hashCode);
-
-            var text = arr[idx].Text;
+            => Add(chars.AsSpan(start, len));
 
-            if (text != null)
-            {
-                var result = arr[idx].Text;
-                if (text.Length == 1 && text[0] == chars)
-                {
-                    return result;
-                }
-            }
-
-            string? shared = FindSharedEntry(chars, hashCode);
-            if (shared != null)
-            {
-                // PERF: the following code does element-wise assignment of a struct
-                //       because current JIT produces better code compared to
-                //       arr[idx] = new Entry(...)
-                arr[idx].HashCode = hashCode;
-                arr[idx].Text = shared;
-
-                return shared;
-            }
-
-            return AddItem(chars, hashCode);
-        }
+        internal unsafe string Add(char chars)
+            => Add(new ReadOnlySpan<char>(&chars, 1));
```

Comment:
I guess we cannot just do `new ROS(ref chars)` due to netframework stuff?

Also, it feels like it would be good to use a parameter name like `@char` here. Though not urgent.

Reply:
i'll check again.  but it seemed to be missing.

Reply:
Yeah, i can't find a nicer way to do this, while still supporting Netfx and other pre-.Net7 targets.